backend/src/main/java/kr/momo/controller/auth/AuthArgumentResolver.java

```diff
@@ -34,15 +36,26 @@ public Long resolveArgument(
             @NonNull NativeWebRequest webRequest,
             WebDataBinderFactory binderFactory
     ) {
-        String token = getToken(webRequest);
+        HttpServletRequest request = (HttpServletRequest) webRequest.getNativeRequest();
+        Cookie[] cookies = request.getCookies();
+        String token = getCookieValue(cookies);
+
+        if (token == null) {
+            throw new MomoException(AuthErrorCode.NOT_FOUND_TOKEN);
+        }
+
         return jwtManager.extract(token);
     }
 
-    private String getToken(NativeWebRequest webRequest) {
-        String header = webRequest.getHeader(AUTHORIZATION);
-        if (header == null) {
-            throw new MomoException(AuthErrorCode.NOT_FOUND_TOKEN);
+    private String getCookieValue(Cookie[] cookies) {
+        if (cookies == null) {
+            return null;
         }
-        return header.replaceFirst(BEARER, "");
+
+        return Arrays.stream(cookies)
+                .filter(cookie -> ACCESS_TOKEN.equals(cookie.getName()))
+                .map(Cookie::getValue)
+                .findFirst()
+                .orElse(null);
     }
```

Comment:
명시적으로 `null`을 다루기보다는 `Optional`  API를 활용해 보면 어떨까요?

```suggestion
        String token = getCookieValue(cookies).orElse("");

        return jwtManager.extract(token);
    }

    private Optional<String> getCookieValue(Cookie[] cookies) {
        if (cookies == null) {
            return Optional.empty();
        }

        return Arrays.stream(cookies)
                .filter(cookie -> ACCESS_TOKEN.equals(cookie.getName()))
                .map(Cookie::getValue)
                .findFirst();
    }
```

Reply:
현재 로직에 따르면 cookies가 null일 때 예외가 발생해야합니다. 
그런데 제안하신 로직에서는 cookies가 null일 때 예외가 발생하지 않고 ""이 반환되는 것 같아요..!

아니면 아래처럼 resolveArgument() 함수 내에서 token의 null 체크 부분을 삭제하고 getCookieValue()함수에서 optional을 사용해서 cookie가 null 값일 때 예외를 발생시키는 방법도 있을 것 같습니다.
```java
    @Override
    public Long resolveArgument(
            @NonNull MethodParameter parameter,
            ModelAndViewContainer mavContainer,
            @NonNull NativeWebRequest webRequest,
            WebDataBinderFactory binderFactory
    ) {
        HttpServletRequest request = (HttpServletRequest) webRequest.getNativeRequest();
        Cookie[] cookies = request.getCookies();
        String token = getCookieValue(cookies);

        return jwtManager.extract(token);
    }

    private String getCookieValue(Cookie[] cookies) {
        cookies = Optional.ofNullable(cookies)
                         .orElseThrow(() -> new MomoException(AuthErrorCode.NOT_FOUND_TOKEN));

        return Arrays.stream(cookies)
                .filter(cookie -> ACCESS_TOKEN.equals(cookie.getName()))
                .map(Cookie::getValue)
                .findFirst()
                .orElseThrow(() -> new MomoException(AuthErrorCode.NOT_FOUND_TOKEN));
    }
```

Reply:
반영하겠습니다 😄

Reply:
@ehBeak 
페드로의 제안은 쿠키의 토큰이 JwtManger의 verifyToken 메서드 한 곳에서 검증 관리되는 게 응집도 측면에서 자연스럽다는 것을 의미하는 것 같습니다.

Reply:
@ehBeak 
페드로의 제안한 의견 맥락은 쿠키의 토큰이 JwtManger의 verifyToken 메서드 한 곳에서 검증 관리되는 게 응집도 측면에서 자연스럽다는 것을 말하고자 하는 것 같습니다.

Reply:
@ehBeak 
페드로 제안의 의미는 쿠키의 토큰이 JwtManger의 verifyToken 메서드 한 곳에서 검증 관리되는 게 응집도 측면에서 자연스럽다는 것을 말하는 것 같습니다.

Reply:
해당 맥락에서 쿠키는 결국 토큰이므로 `null`일 때의 예외 처리 역시 `JwtManager`의 책임이라고 생각했어요🙂